addons/fastroping/functions/fnc_deployRopes.sqf

```diff
@@ -25,6 +25,57 @@ private _hookAttachment = _vehicle getVariable [QGVAR(FRIES), _vehicle];
 {
     private _ropeOrigin = _x;
     private _hook = QGVAR(helper) createVehicle [0, 0, 0];
+    switch (_ropelength) do {
```

Comment:
This could be replaced by a couple array's with `find` and `select`.
But probably not worth he readability loss. But it would get rid of all the duplicate code in here.
Just get the rope length and classname of the item. And the rest of the code is the same for everything